### ISSUE TITLE
Numpy 2.0: `No module named 'numpy.lib.index_tricks'`

### ISSUE THREAD

**Issue by zacharyburnett** [astropy/photutils#1622]
related to [SCSB-85](https://jira.stsci.edu/browse/SCSB-85)

https://github.com/spacetelescope/jwst/actions/runs/6099213054/job/16550865119#step:10:519
```
jwst/outlier_detection/outlier_detection_scaled.py:6: in <module>
    from photutils.aperture import (aperture_photometry, CircularAperture, CircularAnnulus)
.tox/py311-devdeps-xdist/lib/python3.11/site-packages/photutils/__init__.py:16: in <module>
    from . import background
.tox/py311-devdeps-xdist/lib/python3.11/site-packages/photutils/background/__init__.py:7: in <module>
    from .background_2d import *  # noqa: F401, F403
.tox/py311-devdeps-xdist/lib/python3.11/site-packages/photutils/background/background_2d.py:15: in <module>
    from numpy.lib.index_tricks import index_exp
E   ModuleNotFoundError: No module named 'numpy.lib.index_tricks'
```

**Comment by larrybradley** [comment]
This was fixed last night in #1620.

**Comment by zacharyburnett** [comment]
> This was fixed last night in #1621.

Oh sorry! I should have searched for it before making the issue. Thanks!
